### ISSUE TITLE
Feature request: list all functions > limited to only opened file

### ISSUE THREAD

**Issue by marius-sucan** [CoffeeChaton/vscode-autohotkey-NekoHelp-Old#19]
Hello!

The option to list all functions, lists all functions found in any associated file.

Please offer an option to only list the functions from the currently opened file.

Best regards, Marius.

**Comment by CoffeeChaton** [comment]
![img](https://user-images.githubusercontent.com/63182963/215752584-8f2756ea-e9f8-49c8-88ff-218e06a765b4.png)
> Is this what you mean?

It is not difficult to create this option, but it is easy to create different results by opening one less page and one more page. 
If you copy the file or move it to a different folder, opening another toolbar would be a simpler and more stable solution.
I think this is [XY problem](https://en.wikipedia.org/wiki/XY_problem).
What do you really need?
Is it easier to manage a large file by splitting it into multiple smaller files?
Or are you trying to merge things from two different projects together, but you're having problems with the same function names?

Or do you just not need the project to scan certain folders or files?
```jsonc
"AhkNekoHelp.baseScan.IgnoredList": [
  "/\\.",
  ".vscode$",
  "/node_modules$",
  "/ahk_(?:lib|log|music)$",
  "/IMG$",
  "/src$"
]
```

**Comment by marius-sucan** [comment]
Good questions.

The situation is as follows.

I have my project folder with older files of the ahk file I'm working on.  I also have library files.

Now, when I select list all functions, it parses all the files, including the older versions and the library files. That's why I asked if there's a way to limit it to the current/opened file. 

Best regards, Marius.

**Comment by CoffeeChaton** [comment]
I can understand your needs
- I would use CodeLens to add a command at the top of the entire file to list only all the function definitions that I have analyzed for this file. 2.
- `"AhkNekoHelp.baseScan.IgnoredList": ` should fit your needs for now, but I would like to [change the name](https://github.com/CoffeeChaton/vscode-autohotkey-NekoHelp/issues/20) at next version, but the working principle will not change. Later I will briefly explain how this setting works.

**Comment by CoffeeChaton** [comment]
1. When the project starts, return the index file from the root directory of the workspace, and change the [win-style path](https://github.com/CoffeeChaton/vscode-autohotkey-NekoHelp/blob/master/src/tools/fsTools/getUriList.ts#L18) from "\" to "/" , this is to avoid multiple escaping semantics when writing rules. `"\"` in js is like *`* in ahk
2. [Read the setting](https://github.com/CoffeeChaton/vscode-autohotkey-NekoHelp/blob/master/src/tools/fsTools/getUriList.ts#L9), and use [new Regexp()](https://github.com/CoffeeChaton/vscode-autohotkey-NekoHelp/blob/master/src/tools/str2RegexListCheck.ts#L18) to change form setting, and test whether it is passed item by item.
3. When all the tests pass, the file will be added to the project index
4. When the index construction is completed, it will be opened in the background of vscode to parse the file. 

The reason I don't use .gitignore-style glob patterns is because it's 10x slower than pairing with an almost equivalent regexp

In short, if you want to exclude old folders or lib files, add the following some rules to the settings, and use command 1 to rebuild the project index.

```jsonc
    "AhkNekoHelp.baseScan.IgnoredList": [
        "/\\.", // Any folder/file starting with "." contains .git, .vscode, etc.
        "/node_modules$", // file-path Include node_modules
        "/node_modules/", // file-path Include node_modules
        // some demo
        "/neko_Exp_Gdip\\.ahk$",
        "/build(/|$)",
        "/dist(/|$)",
        "/temp(/|$)",
        "/ahk2(/|$)", // file-path Include /ahk2/
        "\\.2\\.ahk$", // file-path Include .2.ahk
        "/oldAhk(/|$)", // file-path Include oldAhk
        "/lib(/|$)" // file-path Include lib
    ],
```

**Comment by marius-sucan** [comment]
Thank you very much ! You can close the issue.

**Comment by CoffeeChaton** [comment]
i move to new repositories  <https://github.com/CoffeeChaton/vscode-autohotkey-NekoHelp-Old/issues/23>
